praw/util/cache.py

```diff
@@ -39,7 +39,7 @@ def __get__(self, obj: Any | None, objtype: Any | None = None) -> Any:
         value = obj.__dict__[self.func.__name__] = self.func(obj)
         return value
 
-    def __init__(self, func: Callable[[Any], Any], doc: str | None = None):
+    def __init__(self, func: Callable[[Any], Any], doc: str | None = None) -> None:
```

Comment:
Can we utilize the built-in cached property?

Reply:
Perhaps? I haven't been looking to make non-type related improvements here.